src/textual/_node_list.py

```diff
@@ -18,8 +18,8 @@ class NodeList:
 
     """
 
-    def __init__(self) -> None:
-        self._nodes: list[DOMNode] = []
+    def __init__(self, nodes: list[DOMNode] = None) -> None:
```

Comment:
I was thinking that being able to optionally provide an initial list of nodes to this constructor could be useful?
I can revert this change if you think it's better to not do this of course :slightly_smiling_face: 

(I checked the [NodeList](https://devdocs.io/dom/nodelist) Web API, but there is no constructor on this one anyway so we cannot do any features comparison here :sweat_smile: )

Reply:
Not required ATM